awx/main/management/commands/precreate_partitions.py

```diff
@@ -0,0 +1,23 @@
+from django.utils.timezone import now
+from django.core.management.base import BaseCommand, CommandParser
+from datetime import timedelta
+from awx.main.utils.common import create_partition
+
+
+class Command(BaseCommand):
+    """Command used to precreate database partitions to avoid pg_dump locks"""
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument('--count', dest='count', action='store', help='The amount of hours of partitions to create', type=int, default=1)
+
+    def _create_partitioned_tables(self, count):
+        start = now()
+        while count > 0:
+            for table in ("main_inventoryupdateevent", "main_jobevent", "main_projectupdateevent", "main_systemjobevent"):
```

Comment:
I would like to make 1 request that we don't hard-code these table names. Import this method instead, you may find it necessary to move into utils:

https://github.com/ansible/awx/blob/f1196fc019313e081d40c8650b14dfe5b3f26898/awx/main/management/commands/cleanup_jobs.py#L22-L23

With that method, you don't need to provide a table name for the loop, but you do need to provide the models. Those would be Job, AdHocCommand, ProjectUpdate, InventoryUpdate, SystemJob.

I listed 5 and you listed 4, I think you're missing the ad hoc commands as well.

Reply:
I refactored based on your feedback @AlanCoding. Let me know if you had something different in mind here.